src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp

```diff
@@ -883,11 +874,27 @@ void CascadiaSettings::CurrentDefaultTerminal(const Model::DefaultTerminal& term
     _currentDefaultTerminal = terminal;
 }
 
+// This function is implicitly called by DefaultTerminals/CurrentDefaultTerminal().
+// It reloads the selection of available, installed terminals and caches them.
+// WinUI requires us that the `SelectedItem` of a collection is member of the list given to `ItemsSource`.
+// It's thus important that _currentDefaultTerminal is a member of _defaultTerminals.
+// Right now this is implicitly the case thanks to DefaultTerminal::Available(),
+// but in the future it might be worthwhile to change the code to use list indices instead.

```

Comment:
I know what you might be thinking... "Create a ticket".
But to be honest I personally wouldn't want to ever fix this issue myself, because it works fine the way it is right now... I just wrote it, to explain why it's safe (thanks to `DefaultTerminal::Available()`).